lib/chloride/host.rb

```diff
@@ -44,7 +45,8 @@ def ssh_connect
         passphrase: @ssh_key_passphrase,
         password: @sudo_password,
         logger: logger,
-        verbose: :warn
+        verbose: :warn,
+        known_hosts: Chloride::SSHKnownHosts.new(@known_hosts_file)
```

Comment:
This line, the implementation of `Chloride::SSHKnownHosts` below, and the dependency bump to `net::ssh` version 4 are the main changes in this PR.